src/flag_gems/ops/conv2d.py

```diff
@@ -498,18 +546,60 @@ def backward(ctx, out_grad):
             device=device,
         )
 
-        grid = lambda meta: (
-            triton.cdiv(in_n * out_height * out_width, meta["BLOCK_NO_HO_WO"]),
+        grid = lambda META: (
+            triton.cdiv(
+                out_grad.shape[0] * input_height * input_width, META["BLOCK_NI_HO_WO"]
+            ),
+            triton.cdiv(int(weight_c), META["BLOCK_CO"]),
             groups,
-            triton.cdiv(weight_c, meta["BLOCK_CI"]),
+        )
+        bias_zero = torch.zeros(groups * weight_c, device=device, dtype=out_grad.dtype)
+        conv2d_forward_kernel[grid](
+            new_out,
+            revert_weight,
+            input_back,
+            bias_zero,
+            out_grad.shape[0],
+            new_out_height,
+            new_out_width,
+            groups * weight_c,
+            input_height,
+            input_width,
+            *new_out.stride(),
+            *revert_weight.stride(),
+            *input_back.stride(),
+            out_c,
+            weight_height,
+            weight_width,
+            1,
+            1,
+            revert_padding_height,
+            revert_padding_width,
+            dilation_height,
+            dilation_width,
+            groups=groups,
         )
 
-        padding_height, padding_width = ctx.padding
-        # return dx,None,None,None,None,None,None
-        conv2d_backward_kernel[grid](
+        weight_back = torch.zeros(
+            out_c * groups,
+            weight_c,
+            weight_height,
+            weight_width,
+            dtype=torch.float32,
+            device=device,
+        )
+
+        grid_weight = lambda meta: (
```

Comment:
since size of weight is generally not large, I suggest not tiling them by BLOCK_CI_HK_WK.